src/RiakClient/Commands/TS/Cell{TValue}.cs

```diff
@@ -0,0 +1,146 @@
+﻿namespace RiakClient.Commands.TS
+{
+    using System;
+    using Messages;
+    using Util;
+
+    public class Cell<TValue> : Cell, IEquatable<Cell<TValue>>
+    {
+        private readonly Type vtype = typeof(TValue);
+        private readonly TValue v;
+
+        public Cell(TValue value)
+            : base(value)
+        {
+            if (ReferenceEquals(null, value))
+            {
+                throw new ArgumentNullException("value", "typed Cells require a non-null value");
+            }
+
+            v = value;
+        }
+
+        public TValue Value
+        {
+            get { return v; }
+        }
+
+        public bool Equals(Cell<TValue> other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return GetHashCode() == other.GetHashCode();
+        }
+
+        public override bool Equals(object obj)
+        {
+            return base.Equals(obj as Cell<TValue>);
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
+        }
+
+        internal override TsCell ToTsCell()
+        {
+            TsCell rv = null;
+
+            if (ReferenceEquals(null, v))
+            {
+                throw new InvalidOperationException("typed Cells require a non-null value");
+            }
+
+            if (vtype == typeof(string))
+            {
+                rv = new TsCell
+                {
+                    varchar_value = RiakString.ToBytes((string)Convert.ChangeType(v, TypeCode.String))
+                };
+            }
+            else if (vtype == typeof(byte[]))
```

Comment:
Converting back into a `Cell<string>` may blow up, so maybe this shouldn't be supported.